### PR TITLE
[feat](inverted index) Adding Storage Format V3 for Inverted Index

### PR DESCRIPTION
**Pull request by zzzxl1993**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

1. "Mainly added the functionality for compressing inverted index position information and dictionary information."

2. "Position information compression must be enabled by setting inverted_index_storage_format to v3 when creating the table."

e.g. 
```
    CREATE TABLE tbl (
          ...
    ) ENGINE=OLAP
    DUPLICATE KEY(`x`)
    COMMENT "OLAP"
    DISTRIBUTED BY RANDOM BUCKETS 1
    PROPERTIES (
    "inverted_index_storage_format" = "V3"
    );
```
4. "The dictionary compression feature requires setting inverted_index_storage_format to v3 and configuring dict_compression to true in the properties."

e.g. 
```
INDEX x_idx (`x`) USING INVERTED PROPERTIES("dict_compression" = "true") COMMENT ''
```

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

File: be/src/olap/rowset/beta_rowset.cpp [hunk 1]
```diff
@@ -703,10 +703,24 @@ Status BetaRowset::show_nested_index_file(rapidjson::Value* rowset_value,
...
+        format_str = "V1";
+        break;
+    case InvertedIndexStorageFormatPB::V2:
+        format_str = "V2";
+        break;
+    case InvertedIndexStorageFormatPB::V3:
+        format_str = "V3";
+        break;
+    default:
+        format_str = "Unknown";
```
Comment (airborne12) [inline be/src/olap/rowset/beta_rowset.cpp:718]: return Error status here

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp [hunk 2]
```diff
@@ -496,7 +496,7 @@ InvertedIndexFileWriter::create_output_stream_v2() {
 
 void InvertedIndexFileWriter::write_version_and_indices_count(lucene::store::IndexOutput* output) {
     // Write the version number
-    output->writeInt(InvertedIndexStorageFormatPB::V2);
+    output->writeInt(_storage_format);
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:499]: may be we can change function name from create_output_stream_v2 to create_output_stream now

File: gensrc/proto/olap_file.proto [hunk 3]
```diff
@@ -343,6 +343,7 @@ enum IndexType {
 enum InvertedIndexStorageFormatPB {
     V1 = 0;
     V2 = 1;
+    V3 = 3;
```
Comment (airborne12) [inline gensrc/proto/olap_file.proto:346]: why 3 not 2?

**Review by airborne12 — COMMENTED** [review]
We need some FE UT test

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by airborne12 — DISMISSED** [review]
LGTM

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by airborne12 — CHANGES_REQUESTED** [review]
Need to add inverted index storage format v3 in CreateReplicaTask.java

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by airborne12** [comment]
run p1

**Comment by airborne12** [comment]
run cloud_p1

**Review by airborne12 — DISMISSED** [review]
LGTM

**Comment by zzzxl1993** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**